test-app/src/main/java/org/readium/r2/testapp/reader/ReaderViewModel.kt

```diff
@@ -272,6 +287,27 @@ class ReaderViewModel(
         activityChannel.send(ActivityCommand.OpenExternalLink(url))
     }
 
+    override fun shouldFollowInternalLink(
+        link: Link,
+        context: HyperlinkNavigator.LinkContext?
+    ): Boolean =
+        when (context) {
+            is HyperlinkNavigator.FootnoteContext -> {
+                val text =
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                        Html.fromHtml(context.content, Html.FROM_HTML_MODE_COMPACT)
+                    } else {
+                        @Suppress("DEPRECATION")
+                        Html.fromHtml(context.content)
+                    }
+
+                val command = VisualFragmentCommand.ShowPopup(text, context.activationPoint)
```

Comment:
I think there's something fishy with the `activationPoint`, it's not lining up with the link when testing on a device.

Reply:
I can restore the previous behaviour until we design a pretty scrollable footnote popup.